[Actions][Docs] Actions and Connectors API Docs (#90974)
elastic/kibana: docs/api/actions-and-connectors/create.asciidoc

[[actions-and-connectors-api-create]]
=== Create action API
++++
<titleabbrev>Create action API</titleabbrev>
++++

> experimental[] Create {kib} actions.

Reviewer: Do we need this `experimental` label?

Author: The [REST API intro](https://kibana_90974.docs-preview.app.elstc.co/guide/en/kibana/master/api.html) says `Each API is experimental and can include breaking changes in any version of Kibana, or might be entirely removed from Kibana.` and all the other APIs said `experimental` so I put that in these API docs as well.

Reviewer: @gchaps need your confirmation about this. Does all REST API should be experimental or it is an obsolete indicator?

Reviewer: According our earlier discussion with team we should keep experimental APIs docs inside README files.

Author: Removed experiemental labels.

[[actions-and-connectors-api-create-request]]
==== Request

`POST <kibana host>:<port>/api/actions/action`

[[actions-and-connectors-api-create-request-body]]
==== Request body

`name`::
  (Required, string) The display name for the action.

`actionTypeId`::
  (Required, string) The action type ID for the action.

`config`::

>   (Required, object) The action configuration. Configuration properties vary depending on

Reviewer 2: ```suggestion
  (Required, object) The configuration for the action. Configuration properties vary depending on
```

>   the action type. Some action types will have no configuration properties. See the specific action type to determine required fields.

Reviewer 2: ```suggestion
  the action type. For information about the configuration properties, refer to <<action-types,Action and connector types>>.
```

`secrets`::

>   (Required, object) The secrets configuration for the action. Secrets properties vary

Reviewer 2: ```suggestion
  (Required, object) The secrets configuration for the action. Secrets configuration properties vary
```

>   depending on the action type. Some action types will have no secrets properties. See the specific ation type to determine required fields.

Reviewer 2: ```suggestion
  depending on the action type. For information about the secrets configuration properties, refer to <<action-types,Action and connector types>>.
```

[[actions-and-connectors-api-create-request-codes]]
==== Response code

`200`::
    Indicates a successful call.

[[actions-and-connectors-api-create-example]]
==== Example

[source,sh]
--------------------------------------------------
$ curl -X POST api/actions/action  -H 'kbn-xsrf: true' -H 'Content-Type: application/json' -d '
{
  "name": "my-action",
  "actionTypeId": ".index",
  "config": {
    "index": "test-index"
  }
}'
--------------------------------------------------
// KIBANA

The API returns the following:

[source,sh]
--------------------------------------------------
{
  "id": "c55b6eb0-6bad-11eb-9f3b-611eebc6c3ad",
  "actionTypeId": ".index",
  "name": "my-action",
  "config": {
    "index": "test-index",
    "refresh": false,
    "executionTimeField": null
  },
  "isPreconfigured": false
}
--------------------------------------------------

> ==== Action type configuration and secrets

Reviewer 2: Since we link to <<action-types,Action and connector types>> above, I would remove lines 70-117.
